ignite/internal/tools/gen-mig-docs/main.go

```diff
@@ -0,0 +1,630 @@
+	"github.com/pkg/errors"
+)
+
+const (
+	igniteCliRepository = "http://github.com/ignite/cli.git"
+	igniteBinaryPath    = "dist/ignite"
+	igniteRepoPath      = "src/github.com/ignite/cli"
+)
+
+var scaffoldCommands = map[string][]string{
+	"chain":  {"chain example --no-module"},
+	"module": {"chain example"},
+	"list": {
+		"chain example",
+		"list list1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"map": {
+		"chain example",
+		"map map1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"single": {
+		"chain example",
+		"single single1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"type": {
+		"chain example",
+		"type type1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"message": {
+		"chain example",
+		"message message1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"query": {
+		"chain example",
+		"query query1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints --module example --yes",
+	},
+	"packet": {
+		"chain example --no-module",
+		"module example --ibc",
+		"packet packet1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+}
+
+var diffExceptions = []string{
+	"*.md",
+	"go.sum",
+	"*_test.go",
+	"*.pb.go",
+	"*.pb.gw.go",
+	"openapi.yml", // FIXME: For some reason removing this file from exceptions causes too much memory usage and OS kills the process
+	".gitignore",
+	"*.html",
+	"*.css",
+	"*.js",
+	"*.ts",
+}
+
+func main() {
+	var logger = log.New(os.Stdout, "", log.LstdFlags)
+
+	fromFlag := flag.String("from", "", "Semver tag to generate migration document from")
+	toFlag := flag.String("to", "", "Semver tag to generate migration document to")
+	flag.Parse()
+
+	var (
+		fromVer, toVer *semver.Version
+		err            error
+	)
+	if fromFlag != nil && *fromFlag != "" {
+		fromVer, err = semver.NewVersion(*fromFlag)
+		if err != nil {
+			logger.Fatalf("Invalid semver tag: %s", *fromFlag)
+		}
+	}
+	if toFlag != nil && *toFlag != "" {
+		toVer, err = semver.NewVersion(*toFlag)
+		if err != nil {
+			logger.Fatalf("Invalid semver tag: %s", *toFlag)
+		}
+	}
+
+	err = run(fromVer, toVer, logger)
+	if err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func run(fromVer, toVer *semver.Version, logger *log.Logger) error {
+	// A temporary directory is created to clone ignite cli repository and build it
+	tmpdir, err := os.MkdirTemp("", ".migdoc")
+	defer os.RemoveAll(tmpdir)
+	if err != nil {
+		return err
+	}
+	logger.Println("Created temporary directory:", tmpdir)
+
+	logger.Println("Cloning", igniteCliRepository)
+	repoDir := filepath.Join(tmpdir, igniteRepoPath)
+	repo, err := git.PlainClone(repoDir, false, &git.CloneOptions{
```

Comment:
just an idea: maybe we can try to download the binaries from the release page. we don't need to build